src/cmdstan/stansummary_helper.hpp

```diff
@@ -190,31 +190,434 @@ int matrix_index(std::vector<int> &index, const std::vector<int> &dims) {
+  // Mean, StdDev, ... percentiles ...
+  std::vector<std::string> sampler_params_header(percentiles.size() + 2);
+  sampler_params_header.at(0) = "Mean";
+  sampler_params_header.at(1) = "StdDev";
+  size_t offset = 2;
+  for (size_t i = 0; i < percentiles.size(); ++i) {
+    sampler_params_header[i + offset] = percentiles[i] + '%';
+  }
+  return sampler_params_header;
+}
+
+std::vector<std::string> get_model_params_header(
+    const std::vector<std::string> &percentiles) {
+  // Mean, MCSE,  StdDev, ... percentiles ..., N_eff, N_eff/s, R_hat
+  std::vector<std::string> model_params_header(percentiles.size() + 6);
+  model_params_header.at(0) = "Mean";
+  model_params_header.at(1) = "MCSE";
+  model_params_header.at(2) = "StdDev";
+  size_t offset = 3;
+  for (size_t i = 0; i < percentiles.size(); ++i) {
+    model_params_header[i + offset] = percentiles[i] + '%';
+  }
+  offset += percentiles.size();
+  model_params_header.at(offset) = "N_Eff";
+  model_params_header.at(offset + 1) = "N_Eff/s";
+  model_params_header.at(offset + 2) = "R_hat";
+  return model_params_header;
+}
+
+void sampler_params_stats(const stan::mcmc::chains<> &chains,
+                          const Eigen::VectorXd &probs,
+                          int sampler_params_start_col,
+                          Eigen::MatrixXd &sampler_params) {
+  sampler_params.setZero();
+  for (int i = 0; i < sampler_params.rows(); ++i) {
+    int i_offset = i + sampler_params_start_col;
+    sampler_params(i, 0) = chains.mean(i_offset);
+    sampler_params(i, 1) = chains.sd(i_offset);
+    Eigen::VectorXd quantiles = chains.quantiles(i_offset, probs);
+    for (int j = 0; j < probs.size(); j++)
+      sampler_params(i, 2 + j) = quantiles(j);
+  }
+}
+
+void model_params_stats(const stan::mcmc::chains<> &chains,
+                        const Eigen::VectorXd &warmup_times,
+                        const Eigen::VectorXd &sampling_times,
+                        const Eigen::VectorXd &probs,
+                        int model_params_start_col,
+                        Eigen::MatrixXd &model_params) {
+  model_params.setZero();
+  double total_warmup_time = warmup_times.sum();
+  double total_sampling_time = sampling_times.sum();
+
+  // Joint log prob lp__
+  model_params(0, 0) = chains.mean(0);
+  double lp_sd = chains.sd(0);
+  double lp_n_eff = chains.effective_sample_size(0);
+  model_params(0, 1) = lp_sd / sqrt(lp_n_eff);
+  model_params(0, 2) = lp_sd;
+  Eigen::VectorXd quantiles = chains.quantiles(0, probs);
+  for (int j = 0; j < probs.size(); j++)
+    model_params(0, 3 + j) = quantiles(j);
+  model_params(0, probs.size() + 3) = lp_n_eff;
+  model_params(0, probs.size() + 4) = lp_n_eff / total_sampling_time;
+  model_params(0, probs.size() + 5) = chains.split_potential_scale_reduction(0);
+
+  // Model parameters
+  for (int i = model_params_start_col; i < chains.num_params(); ++i) {
+    int row_offset = i - model_params_start_col + 1;
```

Comment:
I'm having trouble following why the `row_offset` is dictated by the model parameters starting column index.

Reply:
the model_params_start column == the number of sampler params + `lp__`

how about I change the loop and define loop variables i,j for the row and column index, respectively.
we start filling in row 1 on param_values, starting at column `model_params_start_col`.

Reply:
cleaned up that function - hope it's clearer.